superbench/benchmarks/micro_benchmarks/ib_loopback_performance.py

```diff
@@ -146,33 +150,39 @@ def _preprocess(self):
                     )
                 )
                 return False
-            else:
-                try:
-                    command = os.path.join(self._args.bin_dir, self._bin_name)
-                    numa_cores = get_numa_cores(self._args.numa)
-                    if len(numa_cores) < 2:
-                        self._result.set_return_code(ReturnCode.MICROBENCHMARK_DEVICE_GETTING_FAILURE)
-                        logger.error('Getting numa core devices failure - benchmark: {}.'.format(self._name))
-                        return False
-                    if len(numa_cores) >= 4:
-                        server_core = int(numa_cores[-1])
-                        client_core = int(numa_cores[-3])
-                    else:
-                        server_core = int(numa_cores[-1])
-                        client_core = int(numa_cores[-2])
-                    command += ' ' + str(server_core) + ' ' + str(client_core)
-                    command += ' ' + os.path.join(self._args.bin_dir, self.__support_ib_commands[ib_command])
-                    command += command_mode + ' -F'
-                    command += ' --iters=' + str(self._args.iters)
-                    command += ' -d ' + network.get_ib_devices()[self._args.ib_index].split(':')[0]
-                    command += ' -p ' + str(network.get_free_port())
-                    command += ' -x ' + str(self._args.gid_index)
-                    command += ' --report_gbits'
-                    self._commands.append(command)
-                except BaseException as e:
-                    self._result.set_return_code(ReturnCode.MICROBENCHMARK_DEVICE_GETTING_FAILURE)
-                    logger.error('Getting ib devices failure - benchmark: {}, message: {}.'.format(self._name, str(e)))
-                    return False
+
+            try:
+                self.__sock_fds.append(socket.socket(socket.AF_INET, socket.SOCK_STREAM))
+                # grep SO_REUSE /usr/include/asm-generic/socket.h
+                self.__sock_fds[-1].setsockopt(socket.SOL_SOCKET, getattr(socket, 'SO_REUSEADDR', 2), 1)
```

Comment:
to keep the same as the SO_REUSEPORT line, socket.SO_REUSEPORT does not exist in many python distributions ([for example](https://community.intel.com/t5/Intel-Distribution-for-Python/AttributeError-module-socket-has-no-attribute-SO-REUSEPORT/m-p/1198400)), including conda in current cuda docker image

we can also use socket.SO_REUSEADDR and getattr for SO_REUSEPORT

Reply:
It looks like a bug, not by design based on the feedback from your example. I quickly scan it on Github, and there is no exactly match for your implementation. I don't know whether we need to change it just due to one bug. But it's OK for me.

Reply:
yes it's a workaround for python, because linux kernel supports the SO_REUSEPORT option while python does not
the implementation is based on above [link](community.intel.com/t5/Intel-Distribution-for-Python/AttributeError-module-socket-has-no-attribute-SO-REUSEPORT/m-p/1256539#M1636) and did work (python will pass through the int value to syscall)